### PR TITLE
xyce: 7.6.0 -> 7.7.0

### PR DESCRIPTION
**Pull request by fedeinthemix**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/science/electronics/xyce/default.nix [hunk 1]
```diff
@@ -156,8 +156,10 @@ stdenv.mkDerivation rec {
       "doc/Reference_Guide/Xyce_RG"
       "doc/Release_Notes/Release_Notes_${lib.versions.majorMinor version}/Release_Notes_${lib.versions.majorMinor version}")
 
-    # Release notes refer to an image not in the repo.
-    sed -i -E 's/\\includegraphics\[height=(0.5in)\]\{snllineblubrd\}/\\mbox\{\\rule\{0mm\}\{\1\}\}/' ''${docFiles[2]}.tex
+    # Release notes refer to 4 images not in the repo.
```
Comment (Mindavi) [inline pkgs/applications/science/electronics/xyce/default.nix:159]: Please link an upstream issue here

Comment (fedeinthemix) [reply pkgs/applications/science/electronics/xyce/default.nix:159]: There was a discussion about building the documentation here
https://groups.google.com/g/xyce-users/c/MxeViRo8CT4/m/ppCY7ePLEAAJ
The LaTeX class is not distributed outside of SANDIA. The images are organizations logo which are not distributed freely by organizations.

I'll put a link to the above link.

File: pkgs/applications/science/electronics/xyce/default.nix [hunk 2]
```diff
@@ -32,21 +32,21 @@
...
-  version = "7.6.0";
+  version = "7.7.0";
 
   # useing fetchurl or fetchFromGitHub doesn't include the manuals
   # due to .gitattributes files
   xyce_src = fetchgit {
     url = "https://github.com/Xyce/Xyce.git";
     rev = "Release-${version}";
-    sha256 = "sha256-HYIzmODMWXBuVRZhcC7LntTysuyXN5A9lb2DeCQQtVw=";
+    sha256 = "1p5kyha8wxv2gk0lfpi5j7avg7f4cmcpj8s5gm03v255lzrhwj8p";
```
Comment (pbsds) [inline pkgs/applications/science/electronics/xyce/default.nix:42]: Please use NAR hashes

```suggestion
    hash = "sha256-SOMETHING";
```

File: pkgs/applications/science/electronics/xyce/default.nix [hunk 3]
```diff
@@ -32,21 +32,21 @@
...
   };
 
   regression_src = fetchFromGitHub {
     owner = "Xyce";
     repo = "Xyce_Regression";
     rev = "Release-${version}";
-    sha256 = "sha256-uEoiKpYyHmdK7LZ1UNm2d3Jk8+sCwBwB0TCoHilIh74=";
+    sha256 = "192v4qfhph9za047m1869zhbvfkgzzfhl8j78jp9pp17az8ncg48";
```
Comment (pbsds) [inline pkgs/applications/science/electronics/xyce/default.nix:49]: here too

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 272697` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>xyce</li>
    <li>xyce-parallel</li>
    <li>xyce-parallel.doc</li>
    <li>xyce.doc</li>
  </ul>
</details>

**Pull request merged**